repair/row_level.cc

```diff
@@ -1105,6 +1134,64 @@ class repair_meta {
         cur_rows.push_back(std::move(r));
     }
 
+
+    future<> prepare_sstables_for_incremental_repair() {
+        auto& table = _db.local().find_column_family(_schema->id());
+        table_id tid = table.schema()->id();
+        auto erm = table.get_effective_replication_map();
+        auto& tmap = erm->get_token_metadata_ptr()->tablets().get_tablet_map(tid);
+        auto last_token = _range.end() ? _range.end()->value() : dht::maximum_token();
+        auto id = tmap.get_tablet_id(last_token);
+        auto range = tmap.get_token_range(id);
+        if (range != _range) {
+            on_internal_error(rlogger, format("Repair range={} does not match tablet range={}", _range, range));
+        }
+        auto& tinfo = tmap.get_tablet_info(id);
+        auto sstables_repaired_at = tinfo.sstables_repaired_at;
+        std::vector<compaction_manager::compaction_reenabler> repair_compaction_reenablers;
+        auto sgs = table.storage_groups_for_token_range(range);
+        for (auto& sg : sgs) {
```

Comment:
repair layer shouldn't be aware of storage groups, we can introduce a interface for all the below.

table::start_incremental_repair(range)

which does the awaiting, and takes the lock for each compaction group spanned.